### PR TITLE
Rename and restructure files to conform to Ansible best practice

### PR DESCRIPTION
**Pull request by critzo**
This PR begins the process of refactoring Piecewise's Ansible deployment playbooks and associated files by renaming and restructuring them to conform to Ansible best practices. Note that this PR breaks the application, so it is not to be expected that Piecewise can be deployed from this branch after this PR.

This is mostly a PR moving files only, or renaming them, but also has a couple of changes. In summary, this PR does the following:
- moves all existing playbooks to `/ansible/playbooks`
- creates a stub file for a new deployment playbook: `/ansible/deploy-piecewise.yml`
- moves files from the `/conf` folder into `/ansible/files/` from the master branch and the rhel7 branch, and separates them into three folders: `common`, `debian8` and `rhel7`
- moves the hosts file to `/ansible/inventory/hosts`
- rename `seattle_example` to `local_customizations` to be more location agnostic in naming
- introduces on change to the census block import ansible command
- adds nginx configuration templates for production and development. these came from the author's prototype work, and will be referenced later in subsequent PRs.

---
This change is [Reviewable](https://reviewable.io/reviews/opentechinstitute/piecewise/107)

### CODE REVIEW

File: local_customizations/seattle_tasks.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
 - name: Copy map center to server
   copy:
-    src: center.js
+    src: seattle_example/center.js
```
Comment (nkinkade) [inline local_customizations/seattle_tasks.yml:45]: Should this be `local_customizations/seattle_center.js`?

File: local_customizations/seattle_tasks.yml [hunk 2]
```diff
@@ -75,11 +75,8 @@
 #        /opt/piecewise_web/maryland_blkgrps/cb_2015_24_bg_500k.shp
 #
 - name: Ingest census blocks to postgres
-  command: > 
-    ogr2ogr -f PostgreSQL -t_srs EPSG:4326 -nln seattle_blkgrpce10
-    -nlt MultiPolygon 'PG:user=postgres dbname=piecewise'
-    /opt/piecewise_web/seattle_blkgrpce10/CENSUS10_blkgrp_WGS.shp
-    
+  command: ogr2ogr -f PostgreSQL -t_srs EPSG:4326 -nln seattle_blkgrpce10 -nlt MultiPolygon 'PG:user=postgres dbname=piecewise' /opt/piecewise_web/seattle_blkgrpce10/CENSUS10_blkgrp_WGS.shp
```
Comment (nkinkade) [inline local_customizations/seattle_tasks.yml:78]: Should this point to `/opt/piecewise/local_customizations/geofiles/CENSUS10_blkgrp_WGS.shp`.

What was the reason for removing the "folded scalar" wrapping from the line?

**Review by nkinkade — COMMENTED** [review]
Thanks for this. I added a few small comments, and have one higher level comment. It appears that you have a lot of the same files duplicated between the `debian8` and `rhel7` directories. I would imagine that a lot of those files will be identical between any distro (e.g., collector.ini, nginx.conf, etc.). Are those files going to be different between distros, or could they perhaps be stored in the `ansible/files/common` directory tree to avoid some redundancy?

**Comment by critzo** [comment]
*[local_customizations/seattle_tasks.yml, line 45 at r1](https://reviewable.io:443/reviews/opentechinstitute/piecewise/107#-Kah_X6rWX73Pw3N-dfe-r1-45:-Kaltel0InMoO87tNvzT:b-yikh8s) ([raw file](https://github.com/opentechinstitute/piecewise/blob/e16177134de7b2fbd2738793269a060bb0249fbc/local_customizations/seattle_tasks.yml#L45)):*
<details><summary><i>Previously, nkinkade wrote…</i></summary><blockquote>

Should this be `local_customizations/seattle_center.js`?
</blockquote></details>

It should be, yes, but I'm making an effort to make my PRs smaller and this one is only moving files to new locations and renaming some files.. The correct paths and names will be updated in the next PR in this series.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opentechinstitute/piecewise/107)*

**Comment by critzo** [comment]
*[local_customizations/seattle_tasks.yml, line 78 at r1](https://reviewable.io:443/reviews/opentechinstitute/piecewise/107#-Kah_X6rWX73Pw3N-dfe-r1-78:-Kalthg2dmEw-poS0o3d:ba3uiha) ([raw file](https://github.com/opentechinstitute/piecewise/blob/e16177134de7b2fbd2738793269a060bb0249fbc/local_customizations/seattle_tasks.yml#L78)):*
<details><summary><i>Previously, nkinkade wrote…</i></summary><blockquote>

Should this point to `/opt/piecewise/local_customizations/geofiles/CENSUS10_blkgrp_WGS.shp`.

What was the reason for removing the "folded scalar" wrapping from the line?
</blockquote></details>

This path along with many others will be updated in the next PR. I'm making an effort to make my PRs smaller and this one is only moving files to new locations and renaming some files.

This was an artifact of my testing after the last PR merge when I was experiencing some Ansible failures for that line with the folded scalar. It's my intention to put that back in for readability as a part of the PRs in this series.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opentechinstitute/piecewise/107)*

**Comment by critzo** [comment]
@nkinkade I realized you had a couple other questions related to the system include files in your review summary which didn't appear in Revewiable. 

My approach in adding files from the rhel7 and master branches does add some redundancy because though the contents of some files are the same, their target system locations are different on the two supported architectures. 

The approach I used for system config files was: 
- `/ansible/common/` -- config files that are both the same content and location for all target systems
- `/ansible/rhel7/` -- files that are either specific content or specific location for rhel7 hosts
- `/ansible/debian8/` -- files that are either specific content or specific location for debian8 hosts

I thought that a small bit of redundancy would be better because it reduces the amount of Ansible commands required to place specific files in exact locations depending on the sys== variable, and to encourage readability of the file structure for people who might be doing new deployments and to add support for other OSs potentially.

**Comment by nkinkade** [comment]
I'm wondering if adding a little more complexity to the playbook in order to eliminate the redundancy would be worthwhile? My concern is that having two files with the same content will introduce the possibility for them to get out of sync with one another over time?

For the config files where the content might actually not be the same in small ways due to differing system paths, then there is probably nothing that can be done. However, if there are some files that will be identical no matter which OS they are running on then I would think it could be better to eliminate the redundancy and copying a file into place wouldn't even require `sys==<somesystem>` since that file could be categorically copied into place for every system.

Are there any files where they are pretty much certain to be identical between RHEL and Debian (e.g., pg_hba.conf)?

---

Review status: 0 of 36 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/opentechinstitute/piecewise/107#-:-KamKjnXAH4Cp8keHgC8:b-7x7k9b)*

**Comment by nkinkade** [comment]
With the above said, LGTM to me if you disagree with the above and/or if you feel issues like this will be resolved in future PRs.

---

Review status: 0 of 36 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/opentechinstitute/piecewise/107#-:-KamO9pi9s76I5Sjll3r:b-5ai9qn)*

**Comment by critzo** [comment]
There are certainly a couple of files that are exactly the same content but are required to be put in different locations depending on the target system. There's two strategies for deploying to hosts with different OSs that I see. One is to have a common playbook that checks for the type of system the host is, which is the approach I've taken, using Ansible's conditional "when: sys==<some system>". Another approach might be to have different playbooks for different systems. So one for rhel7, one for debian8, etc., each copying over a common set of files and the files specific to the target host architecture.  

If we were to use a common set of files and not specific sub-directories per architecture, how do you propose keeping state on which files are specific to each architecture? Maybe in the file name.  I like this approach because I can look at the files and see where they are going on the target system type, as well as see the contents and modify them if they need to be different for the target system. And my Ansible playbook is simpler to read, for example:

## Deploy configs for RHEL 7 / Centos systems
- name: Deploy configuration file for RHEL 7 / Centos systems
  copy:
    remote_src: true
    src: "files/rhel7/"
    dest: /
    owner: root
    group: root
  when: sys == "rhel7"

I'd be interested in continuing the discussion, as both approaches accomplish the same thing. I'm not so worried about the files that might be the same getting out of sync and the small bit of redundancy because I'm balancing it with the need to structure the application so it's easy to understand what's happening.  I'll mark this resolved and proceed and we can discuss further, perhaps with others weighing in. 

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/opentechinstitute/piecewise/107#-:-KamP6-a-9eBNny0ONi-:b-7bjt5v)*

**Pull request merged**